### PR TITLE
meta, util: add unit tests

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Add unit tests.
Before this PR:
ok github.com/pingcap/tidb/meta	0.033s	coverage: 76.9% of statements
Now:
ok  	github.com/pingcap/tidb/meta	0.040s	coverage: 85.7% of statements

### What is changed and how it works?
Add unit tests and tiny clean up.
Remove some useless functions like `parseDatabaseID `, `parseTableID `.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @tiancaiamao @XuHuaiyu

File: meta/meta.go [hunk 1]
```diff
@@ -196,52 +175,34 @@ func (m *Meta) GenSchemaVersion() (int64, error) {
...
-	if err != nil {
-		return errors.Trace(err)
-	} else if v == nil {
-		return ErrDBNotExists
+	if err == nil && v == nil {
+		err = ErrDBNotExists
 	}
-
-	return nil
+	return errors.Trace(err)
```
Comment (zz-jason) [inline meta/meta.go]: no need to trace

File: meta/meta.go [hunk 2]
```diff
@@ -196,52 +175,34 @@ func (m *Meta) GenSchemaVersion() (int64, error) {
...
+	if err == nil && v != nil {
+		err = ErrDBExists
 	}
-
-	if v != nil {
-		return ErrDBExists
-	}
-
-	return nil
+	return errors.Trace(err)
```
Comment (zz-jason) [inline meta/meta.go]: ditto

File: meta/meta.go (hunk 1 above)
Comment (zimulala) [reply meta/meta.go]: It's original code,  and other cases in this file all use the trace. I think removing trace can be in other PR.

**Pull request merged**